packages/store/test/StoreCoreDynamic.t.sol

```diff
@@ -96,41 +98,43 @@ contract StoreCoreDynamicTest is Test, GasReporter, StoreMock {
     // Expect a StoreSpliceRecord event to be emitted
     vm.expectEmit(true, true, true, true);
     emit Store_SpliceDynamicData(
-      _tableId,
-      _keyTuple,
+      tableId,
+      keyTuple,
       uint48(secondDataBytes.length - byteLengthToPop),
       uint40(byteLengthToPop),
       new bytes(0),
-      PackedCounterLib.pack(newDataBytes.length, thirdDataBytes.length).unwrap()
+      PackedCounterLib.pack(newDataBytes.length, thirdDataBytes.length)
     );
 
     // Pop from second field
     startGasReport("pop from field (cold, 1 slot, 1 uint32 item)");
-    StoreCore.popFromDynamicField(_tableId, _keyTuple, 0, byteLengthToPop);
+    StoreCore.popFromDynamicField(tableId, keyTuple, 0, byteLengthToPop);
     endGasReport();
     // Get second field
-    bytes memory loadedData = StoreCore.getField(_tableId, _keyTuple, 1, fieldLayout);
+    bytes memory loadedData = StoreCore.getField(tableId, keyTuple, 1, fieldLayout);
     // Verify loaded data is correct
     assertEq(loadedData, newDataBytes);
 
     // Reset the second field and pop again (but warm this time)
-    StoreCore.setField(_tableId, _keyTuple, 1, dataBytes, fieldLayout);
+    StoreCore.setField(tableId, keyTuple, 1, dataBytes, fieldLayout);
     startGasReport("pop from field (warm, 1 slot, 1 uint32 item)");
-    StoreCore.popFromDynamicField(_tableId, _keyTuple, 0, byteLengthToPop);
+    StoreCore.popFromDynamicField(tableId, keyTuple, 0, byteLengthToPop);
     endGasReport();
     // Get second field
-    loadedData = StoreCore.getField(_tableId, _keyTuple, 1, fieldLayout);
+    loadedData = StoreCore.getField(tableId, keyTuple, 1, fieldLayout);
     // Verify loaded data is correct
     assertEq(loadedData, newDataBytes);
 
     // Verify none of the other fields were impacted
-    assertEq(bytes32(StoreCore.getField(_tableId, _keyTuple, 0, fieldLayout)), firstDataBytes);
-    assertEq(StoreCore.getField(_tableId, _keyTuple, 2, fieldLayout), thirdDataBytes);
+    assertEq(bytes32(StoreCore.getField(tableId, keyTuple, 0, fieldLayout)), firstDataBytes);
+    assertEq(StoreCore.getField(tableId, keyTuple, 2, fieldLayout), thirdDataBytes);
   }
 
   function testPopFromThirdField() public {
     FieldLayout fieldLayout = StoreCore.getFieldLayout(_tableId);
     bytes memory dataBytes = thirdDataBytes;
+    ResourceId tableId = _tableId;
+    bytes32[] memory keyTuple = _keyTuple;
```

Comment:
we could use immutable too, it's about as convenient as storage, but without the gas overhead. It's only useful for constant-like values tho